gatorgrade/output/output.py

```diff
@@ -0,0 +1,127 @@
+"""Run checks and display whether each has passed or failed."""
+
+import subprocess
+from pathlib import Path
+from typing import List, Union
+import gator
+import rich
+from gatorgrade.input.checks import ShellCheck, GatorGraderCheck
+from gatorgrade.output.check_result import CheckResult
+
+# Disable rich's default highlight to stop number coloring
+rich.reconfigure(highlight=False)
+
+
+def _run_shell_check(check: ShellCheck) -> CheckResult:
+    """Run a shell check.
+
+    Args:
+        check: The shell check to run.
+
+    Returns:
+        The result of running the shell check as a CheckResult.
+    """
+    result = subprocess.run(
+        check.command,
+        shell=True,
+        check=False,
+        timeout=300,
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+    )
+    passed = result.returncode == 0
+    diagnostic = "" if passed else f'The command "{check.command}" failed'
```

Comment:
In GatorGradle, this is the stdout+stderr of the command run, if that has any content after trimming -- if not, it is exactly this.